aipl/interpreter.py

```diff
@@ -158,13 +158,7 @@ def run_cmdlist(self, cmds:List[Command], inputs):
         return inputs
 
     def call_cmd(self, cmd:Command, contexts:List[Mapping], *inputs):
-        try:
-            ret = cmd.op(self, *inputs, *fmtargs(cmd.args, contexts), **fmtkwargs(cmd.kwargs, contexts))
-        except Exception as e:
```

Comment:
We need a way to manage and continue errors during processing.  For instance if a small number of links raise an exception when fetching, we still want to process the 99% of links that are fine.

Reply:
Is it enough to make sure `debug` mode is on when the tests are run?

Reply:
> This was already merged, I forgot to link it in the issue: #8

Ah, I didn't see that, sorry! But, yes, being able to run everything locally with `pytest`, `pytest --lf` will work, etc.

> **Edit** Ah, I see, so that `pytest .` just runs everything. If we do go with this approach (I do think it is better), then we should make sure to remove [Run aipl tests](https://github.com/saulpw/aipl/blob/35ecc94f4946ba0c73113c6bdd6b99b4e62d5453/.github/workflows/main.yml#L37) and I think add a `aipl --help` to the CI (to test that the bin startup works).

Sounds reasonable.

The other approach would be to run from pytest but still shell out to a to the aipl cli.

At the moment I'd lean towards the tests as I have them here, and some basic cli tests as you suggest.

Reply:
> We need a way to manage and continue errors during processing. For instance if a small number of links raise an exception when fetching, we still want to process the 99% of links that are fine.

I think I'd rather find something more explicit than just dropping errors though. We should be able to represent a result type pretty easily for operations that are expected to fail. Something like:

[{'error': 'http refused 404', 'ok': ''}, {'error': '', 'ok': 'data...'}]

A !filter-ok or just rely on the last value thing?

> Is it enough to make sure self.debug mode is on when the tests are run?

For now possibly, but it still means that the exit code is wrong for scripts with assertions.

At least could we add

```
except AssertionError:
    raise
````